### PR TITLE
Review

### PR DESCRIPTION
**Pull request by popaula937**
Changes involve the refactoring of hypothesis/model configuration and how they are used.

### CODE REVIEW

File: examples/main_refactor.py [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+from tvb_epilepsy.base.constants import FOLDER_RES, FOLDER_FIGURES, SAVE_FLAG, SHOW_FLAG, SIMULATION_MODE, \
+    TVB, DATA_MODE
+from tvb_epilepsy.base.disease_hypothesis import DiseaseHypothesis
+from tvb_epilepsy.base.equilibrum_service import EquilibrumComputationService
+from tvb_epilepsy.base.plot_tools import plot_nullclines_eq, plot_sim_results
+from tvb_epilepsy.base.utils import initialize_logger, set_time_scales, calculate_projection, filter_data
+from tvb_epilepsy.custom.read_write import write_h5_model, write_ts_epi, write_ts_seeg_epi
+from tvb_epilepsy.tvb_api.epileptor_models import EpileptorDP
+
+if DATA_MODE is TVB:
```
Comment (maedoc) [inline examples/main_refactor.py]: If `DATA_MODE` is a constant, why is it in a `if`-`else`? Better to use a command line flag for this.

File: examples/main_refactor.py [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+    hpf_flag = False
+    hpf_low = max(16.0, 1000.0 / time_length)  # msec
+    hpf_high = min(250.0, fsAVG)
+
+    if status:
+
+        model = simulator_instance.model
+
+        logger.info("\nSimulated signal return shape: " + str(tavg_data.shape))
+        logger.info("Time: " + str(scale_time * ttavg[0]) + " - " + str(scale_time * ttavg[-1]))
```
Comment (maedoc) [inline examples/main_refactor.py]:
```python
logger.info("Time: %s - %s", scale_time * ttavg[0], scale_time * ttavg[-1])
```

File: examples/main_refactor.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+        logger.info("Values: " + str(tavg_data.min()) + " - " + str(tavg_data.max()))
+
+        write_h5_model(simulator_instance.prepare_for_h5(), folder_name=data_folder, file_name=hypothesis.get_name() + "_sim_settings.h5")
+
+        # Pack results into a dictionary, high pass filter, and compute SEEG
+        res = dict()
+        time = scale_time * numpy.array(ttavg, dtype='float32')
+        dt = numpy.min(numpy.diff(time))
+        for iv in range(len(vois)):
```
Comment (maedoc) [inline examples/main_refactor.py]:
```python
for iv, voi in enumerate(vois):
```

File: examples/main_refactor.py [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+            res[vois[iv]] = numpy.array(tavg_data[:, iv, :], dtype='float32')
```
Comment (maedoc) [inline examples/main_refactor.py]:
```python
for iv, voi in enumerate(vois):
    res[voi] = tavg_data[:, iv, :].astype('f')
```

File: examples/main_refactor.py [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+        if model._ui_name == "EpileptorDP2D":
```
Comment (maedoc) [inline examples/main_refactor.py]: private attr access

File: examples/main_refactor.py [hunk 6]
```diff
@@ -0,0 +1,165 @@
...
+            raw_data = numpy.dstack([res["x1"], res["z"], res["x1"]])
+            lfp_data = res["x1"]
+            for i in range(len(projections)):
```
Comment (maedoc) [inline examples/main_refactor.py]: as before
```python
for i, proj in enumerate(projections):
    res['seeg%d' % i] = res['z'].dot(proj.T)
```

File: examples/main_refactor.py [hunk 7]
```diff
@@ -0,0 +1,165 @@
...
+
+        write_ts_epi(raw_data, dt, lfp_data, path=os.path.join(FOLDER_RES, hypothesis.name + "_ep_ts.h5"))
+        del raw_data, lfp_data
+
+        for i in range(len(projections)):
+            write_ts_seeg_epi(res['seeg' + str(i)], dt, path=os.path.join(FOLDER_RES, hypothesis.name + "_ep_ts.h5"))
+
+        res['time'] = time
+
+        del ttavg, tavg_data
```
Comment (maedoc) [inline examples/main_refactor.py]: restructure this main clause as a set of small functions and you won't need this.  you should never need `del` except at a module global scope.

File: examples/main_refactor.py [hunk 8]
```diff
@@ -0,0 +1,165 @@
...
+
+        # Plot results
+        plot_nullclines_eq(model_configuration, head.connectivity.region_labels,
+                           special_idx=numpy.concatenate((lsa_hypothesis.get_x0_indices(),
```
Comment (maedoc) [inline examples/main_refactor.py]: why aren't you using `property`s here, e.g. `lsa.x0_indices`?

Comment (maedoc) [inline examples/main_refactor.py]: `np.r_[a, b]` concatenates vectors `a` and `b` and is idiomatic for simple cases

File: tvb_epilepsy/base/disease_hypothesis.py [hunk 9]
```diff
@@ -0,0 +1,92 @@
+# coding=utf-8
+"""
+Hypothesis POJO class.
```
Comment (maedoc) [inline tvb_epilepsy/base/disease_hypothesis.py]: POJO is Java terminology

File: examples/main_refactor.py [hunk 10]
```diff
@@ -0,0 +1,165 @@
...
+    x0_values = numpy.random.normal(0.85, 0.02, (len(x0_indices),))
+
+    hypothesis = DiseaseHypothesis("x0", head.connectivity, x0_values, x0_indices, [], [], [], "x0_Hypothesis")
+
+    all_regions_one = numpy.ones((hypothesis.get_number_of_regions(),), dtype=numpy.float32)
+
+    epileptor_model = EpileptorDP(Iext1=3.1 * all_regions_one, yc=all_regions_one,
+                                  K=10 * all_regions_one / hypothesis.get_number_of_regions())
+
+    equilibrum_service = EquilibrumComputationService(hypothesis, epileptor_model)
```
Comment (maedoc) [inline examples/main_refactor.py]: couldn't this just be named `Equilibrium`?

File: examples/main_refactor.py [hunk 11]
```diff
@@ -0,0 +1,165 @@
...
+    fs = 2 * 4096.0
+    scale_time = 2.0
+    time_length = 3000.0
+    scale_fsavg = 2.0
+    report_every_n_monitor_steps = 10.0
+    (dt, fsAVG, sim_length, monitor_period, n_report_blocks) = set_time_scales(fs=fs, dt=None,
+                                                                               time_length=time_length,
+                                                                               scale_time=scale_time,
+                                                                               scale_fsavg=scale_fsavg,
+                                                                               report_every_n_monitor_steps=report_every_n_monitor_steps)
```
Comment (maedoc) [inline examples/main_refactor.py]: this line doesn't fit in the github review window.. way too long.  please adhere to PEP8: PyCharm supports it, and anyone reading your code expects it.

File: examples/main_refactor.py [hunk 12]
```diff
@@ -0,0 +1,165 @@
+"""
```
Comment (liadomide) [inline examples/main_refactor.py]: This file could be removed, and its content directly replace the original file main_vep.py.
This started as a clone of main_vep.py

**Comment by liadomide** [comment]
We still have some changes that we could do, but this state is already an improvement for the architecture of the module (by splitting between Hypothesis IN | services | Configutions OUT), thus we should merge.

**Pull request merged**